src/value/borrowed.rs

```diff
@@ -137,6 +138,32 @@ impl<'val> SymbolToken for BorrowedSymbolToken<'val> {
     }
 }
 
+/// An borrowed implementation of [`Builder`].
+impl<'val> Builder for BorrowedElement<'val> {
+    type Element = BorrowedElement<'val>;
+    type Sequence = BorrowedSequence<'val>;
+
+    fn new_null(e_type: IonType) -> Self::Element {
+        BorrowedValue::Null(e_type).into()
+    }
+
+    fn new_clob(bytes: &'static [u8]) -> Self::Element {
+        BorrowedValue::Clob(bytes).into()
+    }
+
+    fn new_blob(bytes: &'static [u8]) -> Self::Element {
```

Comment:
I was a little surprised to see the `new_*lob` functions requiring a `'static` lifetime, but I see that's caused by `Self::Element` not having a lifetime of its own. My kingdom for a GATs! 😞